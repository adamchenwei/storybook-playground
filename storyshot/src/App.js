import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: 'Adam Chen Wei',
      something: props.something,
    };
  }

  asyncCallResult() {
    // axios('').then((data) => {
    //   console.log(data);
    // })
  }

  render() {
    return (
      <div className="App">
        {/* Async Call */}
        {/* State */}
        {/* Child Component */}
        {/* list.map */}
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
          <h2>{this.state.something}</h2>
        </header>

        <p className="App-intro">
          To get started22, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;
