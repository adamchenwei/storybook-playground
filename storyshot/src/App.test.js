import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

describe('App Component', () => {
  test('renders as expected', () => {
    const tree = renderer.create(
      <App />
    ).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
})