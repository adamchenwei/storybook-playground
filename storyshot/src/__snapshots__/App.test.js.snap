// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders as expected 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="IMAGE_MOCK"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h2 />
  </header>
  <p
    className="App-intro"
  >
    To get started22, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
</div>
`;
